### ISSUE TITLE
Allow a LED range for ws2812

### ISSUE THREAD

**Issue by pinoyyid** [arendst/Sonoff-Tasmota#4447]
**Have you look for this feature in other issues and in the wiki?**
Yes
**Is your feature request related to a problem? Please describe.**
I have a 65 pixel ws2812 linked to a nodeMCU as a headboard. Logic in my controller divides this into 5 banks of 13 pixels. If I want banks 1,2 & 3 in blue, and banks 4 and 5 in red. I send 5 backlog commands...
```
backlog led1 0000ff; led2 0000ff ... led13 0000ff
backlog led14 0000ff; led15 0000ff ... led26 0000ff
...
backlog led52 ff0000; led53 ff0000 ... led65 ff0000
```
What I'm finding is that this seems to fail unless I throttle the messages. I think this is a bug, so I'll raise a separate issue about that. In the meantime, this feature request is ...

**Describe the solution you'd like**
Could the `LEDx rrggbb` command support a range, eg. `LED1,3,5,7-10 rrggbb`. This would simplify my multiple backlogs into a two commands, `LED1-39 0000ff ; LED40-65 ff0000`

**Describe alternatives you've considered**
backlog, but this seems to cause a separate issue

**Comment by ascillato** [comment]
> Could the LEDx rrggbb command support a range

You can use the rules feature instead and make the led range there using EVENT command. Please read the wiki at rules and also check the examples there.

> What I'm finding is that this seems to fail unless I throttle the messages. I think this is a bug

Sorry, Tasmota has a limitation in the amount of characters. Not enough memory for more characters to be processed at once. You need to send separated messages.

**Comment by ascillato2** [comment]
Closing this issue as it has been answered.

### Support Information
See [Wiki](https://github.com/arendst/Sonoff-Tasmota/wiki) for more information.
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.

**Comment by pinoyyid** [comment]
I'll look at Rules to see if that provides a workaround.
Please note, this issue is a feature request, not a question.
Feel free to consider it and then reject it as "won't implement", but please don't close it as "question answered".

**Comment by ascillato** [comment]
No problem at all.
Rules feature was added by Theo to provide new behaviours for Tasmota like the one you wrote.
If you need any help or support with that, please just ask or find us in the chat.

**Comment by pinoyyid** [comment]
I've looked at Rules and they will not work for my use case. Please can you reopen the issue and consider whether or not the feature should be implemented.

**Comment by ascillato** [comment]
Rule example for this case:

go to the console and type:

```
rule1 1
rule1 on event#block1 do backlog led1 %value%; led2 %value%; led3 %value% endon on event#block2 do backlog led4 %value%; led5 %value%; led6 %value% endon
```

Then by commands in the console or by MQTT or by HTTP you can send the command:

* Console:

`event block1=0000ff`

* MQTT:

`cmnd/your_sonoff_topic/event block1=0000ff`

* HTTP:

`http://sonoff_IP/cm?cmnd=event%20block1=0000ff `

Please, tell us if this fit your requirements. Thanks.

**Comment by ascillato** [comment]
As requested was added the label enhancement looking for someone to design and implement your modification and to make a PR.

--------------------------------

FYI

### Tasmota's Rules Feature Clarification

Rules are 3 SETs of 511 characters each one. You can put all the rules together in the same rule set. The example provided has 2 events (event block1 and event block2) in the same rule set. So, you can make as many rules and events as you want up to 1533 characters.

**Comment by ascillato** [comment]
The only drawback of the rules' example proposed is that when you do the command `event block1=0000ff`, the leds will change color one by one with a 20 ms delay between them. Funny effect :) But they are not going to change ALL at the same time.

Have you tried the example?

**Comment by ascillato** [comment]
As requested was added the label enhancement looking for someone to design and implement your modification and to make a PR.

--------------------------------

FYI

### Tasmota's Rules Feature Clarification

Rules are 3 SETs of 511 characters each one. You can put all the rules together in the same rule set. The example provided has 2 events (event block1 and event block2) in the same rule set. So, you can make as many rules and events as you want up to 1533 characters.

**Comment by ascillato** [comment]
Anyway, have you tried the example?

**Comment by pinoyyid** [comment]
I haven't. It would require a lot of rework to my application and be a brittle solution at best. For now, my workaround is to simply throttle the backlog commands sufficiently to not overrun the command buffer.

**Comment by ascillato** [comment]
Ok, Great!

So you were doing critisism over a rule that you never tested to know it it works or not!!!

Nice!!!

**Comment by ascillato2** [comment]
brittle solution, kludge suggestion Not the best words asking for help

**Comment by ascillato2** [comment]
I will moderate this chat. 

You idea is ok, don't worry about that. but next time be more kind in open source community !!!!!

**Comment by pinoyyid** [comment]
Thanks for considering the enhancement request. 
Can I also take this opportunity to congratulate Theo and the community for what, imho, is the best structured and presented open source project I have ever dealt with. 

I lack the c++ and MCU expertise to contribute quality PR's to this project, but I can make considered enhancement suggestions that will benefit a range of use cases and thus help move Tasmota ever onwards and upwards.

**Comment by ascillato2** [comment]
Nice idea. Added your request to the list of requested effects in the already open issue #3480

Thanks for sharing your ideas.

**Comment by germil** [comment]
> Rule example for this case:
> 
> go to the console and type:
> 
> ```
> rule1 1
> rule1 on event#block1 do backlog led1 %value%; led2 %value%; led3 %value% endon on event#block2 do backlog led4 %value%; led5 %value%; led6 %value% endon
> ```
> Then by commands in the console or by MQTT or by HTTP you can send the command:
> 
> * by Console:
> 
> `event block1=0000ff`
> `event block2=ff0000`
> 
> * by MQTT:
> 
> `cmnd/your_sonoff_topic/event block1=0000ff`
> `cmnd/your_sonoff_topic/event block2=ff0000`
> 
> _Note:_
> _MQTT MESSAGE:_ `cmnd/your_sonoff_topic/event block1`
> _PAYLOAD:_ `0000ff`
> 
> * by HTTP:
> 
> `http://sonoff_IP/cm?cmnd=event%20block1=0000ff `
> `http://sonoff_IP/cm?cmnd=event%20block2=ff0000 `
> 
> Please, tell us if this fit your requirements. Thanks.

Hello,
thats exactly what i want. I can get it to work out from the command line in tasmota, but cant get it to work with mqtt command from node red.
I use that flow:
[{"id":"d16b4eb1.dd64d","type":"tab","label":"Flow 1","disabled":false,"info":""},{"id":"6a3fe74d.bc29e8","type":"inject","z":"d16b4eb1.dd64d","name":"","topic":"","payload":"684A52","payloadType":"str","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":310,"y":200,"wires":[["b93017d9.e5fa38","d6f2c63b.b5a6f8","fe8b8ed8.11569","87d77c48.7abf8"]]},{"id":"b93017d9.e5fa38","type":"mqtt out","z":"d16b4eb1.dd64d","name":"","topic":"cmnd/mi2smart7/event block1","qos":"","retain":"","broker":"74485cce.d3f9f4","x":630,"y":200,"wires":[]},{"id":"d6f2c63b.b5a6f8","type":"mqtt out","z":"d16b4eb1.dd64d","name":"","topic":"cmnd/mi2smart7/event block2","qos":"","retain":"","broker":"74485cce.d3f9f4","x":630,"y":260,"wires":[]},{"id":"fe8b8ed8.11569","type":"mqtt out","z":"d16b4eb1.dd64d","name":"","topic":"cmnd/mi2smart7/event block3","qos":"","retain":"","broker":"74485cce.d3f9f4","x":630,"y":320,"wires":[]},{"id":"87d77c48.7abf8","type":"mqtt out","z":"d16b4eb1.dd64d","name":"","topic":"cmnd/mi2smart7/event block4","qos":"","retain":"","broker":"74485cce.d3f9f4","x":630,"y":380,"wires":[]},{"id":"74485cce.d3f9f4","type":"mqtt-broker","z":"","name":"","broker":"localhost","port":"1883","clientid":"","usetls":false,"compatmode":true,"keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","closeTopic":"","closePayload":"","willTopic":"","willQos":"0","willPayload":""}]

![image](https://user-images.githubusercontent.com/25988277/50602820-bb42cf80-0eb8-11e9-8259-a4e970dbccd1.png)

Thanks in advance

**Comment by moonlitknight** [comment]
My request was merged into #3480 which is sad because my request is very different from a generic "I want more ws2812 effects".

However, it seems that what I was requesting *is* possible, but not documented. The LED command supports a contiguous range thus: `LED1 00FFFF 00FFFF 00FFFF` which is a direct equivalent of my request for `LED1-3 00FFFF` and solves my problem completely.
